### PR TITLE
Document HTTP redirections follow options

### PR DESCRIPTION
**Pull request by guillempages**
Update documentation according to PR https://github.com/esphome/esphome/pull/3100

## Description:
Add force_redirect and redirect_limit options

**Related issue (if applicable):**

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3100

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by oxan — CHANGES_REQUESTED** [review]
Can you rebase this upon the `next` branch? Right now your PR contains all changes from the `current` branch as well, which makes it hard to review.

**Comment by guillempages** [comment]
Oops. Sorry :-(
I did that from the UI, and saw that I was pushing to the wrong branch, so I changed that, but I missed the fact that I had already branched off the wrong branch.
Fixed now.

File: components/http_request.rst [hunk 1]
```diff
@@ -22,6 +22,8 @@ Configuration variables:
 - **useragent** (*Optional*, string): User-Agent header for requests. Defaults to ``ESPHome``.
 - **timeout** (*Optional*, :ref:`config-time`): Timeout for request. Defaults to ``5s``.
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **force_redirects** (*Optional*, boolean): Enable HTTP forced redirection. Defaults to ``true``.
+- **redirect_limit** (*Optional*, integer): Maximum amount of redirects to follow when enabled.
```
Comment (guillempages) [inline components/http_request.rst]:
```suggestion
- **redirect_limit** (*Optional*, integer): Maximum amount of redirects to follow when enabled. Defaults to ``3``.
```

File: components/http_request.rst [hunk 2]
```diff
@@ -22,6 +22,8 @@ Configuration variables:
 - **useragent** (*Optional*, string): User-Agent header for requests. Defaults to ``ESPHome``.
 - **timeout** (*Optional*, :ref:`config-time`): Timeout for request. Defaults to ``5s``.
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **force_redirects** (*Optional*, boolean): Enable HTTP forced redirection. Defaults to ``true``.
```
Comment (oxan) [inline components/http_request.rst]: Force -> follow:
```suggestion
- **follow_redirects** (*Optional*, boolean): Enable following HTTP redirects. Defaults to ``true``.
```

**Review by oxan — CHANGES_REQUESTED** [review]
No worries!

File: components/http_request.rst (hunk 2 above)
Comment (guillempages) [reply components/http_request.rst]: (facepalm)

**Review by oxan — APPROVED** [review]

**Pull request merged**